### ISSUE TITLE
Seurat3.0_"FIt-SNE"_ERROR

### ISSUE THREAD

**Issue by ChevyXu** [satijalab/seurat#1030]
Dear Developers,, wanna ask for your help, An ERROR occurred when I try to run TSNE in Seurat release 3.0 in **FIt-SNE** mode:
```R
> source('D:/Software/Flt-SNE/FIt-SNE-master/fast_tsne.R', chdir=T)
FIt-SNE R wrapper loading.
FIt-SNE root directory was set to D:/Software/Flt-SNE/FIt-SNE-master
> pbmc <- RunTSNE(object = pbmc, 
+                reduction.use = "pca", 
+                dims.use = 1:15, 
+                tsne.method = "FIt-SNE", 
+                nthreads = 4, 
+                reduction.name = "FItSNE", 
+                reduction.key = "FItSNE_", 
+                fast_tsne_path = "D:/Software/Flt-SNE/FIt-SNE-master/bin/FItSNE.exe", 
+                max_iter = 2000)
=============== t-SNE v1.0.0 ===============
fast_tsne data_path: C:\Users\DELL\AppData\Local\Temp\Rtmpsjeisc\fftRtsne_data_3f70566d1ade.dat
fast_tsne result_path: C:\Users\DELL\AppData\Local\Temp\Rtmpsjeisc\fftRtsne_result_3f70661a7a5f.dat
fast_tsne nthreads: 4
Read the following parameters:
	 n 10050 by d 5 dataset, theta 0.500000,
	 perplexity 30.000000, no_dims 2, max_iter 2000,
	 stop_lying_iter 250, mom_switch_iter -1,
	 momentum -30.000000, final_momentum 0.000000,
	 learning_rate 12.000000, K 0, sigma 0.000000, nbody_algo -1,
	 knn_algo 0, early_exag_coeff 0.000000,
	 no_momentum_during_exag 0, n_trees 1072693248, search_k 50,
	 start_late_exag_iter -675358355, late_exag_coeff 0.000000
	 nterms 1075668789, interval_per_integer 1.318504, min_num_intervals -566521812
Read the 10050 x 5 data matrix successfully. X[0,0] = -0.000000
Using random seed: 0
Will use momentum during exaggeration phase
Computing input similarities...
Using perplexity, so normalizing input data (to prevent numerical problems)
Using perplexity, not the manually set kernel width.  K (number of nearest neighbors) and sigma (bandwidth) parameters are going to be ignored.
Invalid knn_algo param
Error in fftRtsne(X = object, dims = dim.embed, rand_seed = seed.use,  : 
  tsne call failed
```

Could you show me that what is the correct way to runTSNE in FIt-SNE mode in Seurat3.
The Flt-SNE was downloaded from [Github KlugerLab/FIt-SNE](https://github.com/KlugerLab/FIt-SNE)

Thanks,
Best wishes

**Comment by timoast** [comment]
Hi,

Please note that several argument names have changed between Seurat v2 and v3. Here you're using the old argument names `reduction.use` and `dims.use`, replaced by `reduction` and `dims` in v3. Consequently they're not being set properly and you're actually running tSNE using only the first 5 PCs  rather than the 15 you requested (you can see in the FIt-SNE call, "n 10050 by d 5 dataset").

However, I don't think that is the cause of this error. Can you confirm you are running the most current version of Seurat v3? We saw a similar issue (#904) after FIt-SNE updated to v1, and this was fixed in commit ae9d28e91b9bb6b750c36c3fb4783fef7e3fb976 

Tim

**Comment by ChevyXu** [comment]
> Hi,
> 
> Please note that several argument names have changed between Seurat v2 and v3. Here you're using the old argument names `reduction.use` and `dims.use`, replaced by `reduction` and `dims` in v3. Consequently they're not being set properly and you're actually running tSNE using only the first 5 PCs rather than the 15 you requested (you can see in the FIt-SNE call, "n 10050 by d 5 dataset").
> 
> However, I don't think that is the cause of this error. Can you confirm you are running the most current version of Seurat v3? We saw a similar issue (#904) after FIt-SNE updated to v1, and this was fixed in commit [ae9d28e](https://github.com/satijalab/seurat/commit/ae9d28e91b9bb6b750c36c3fb4783fef7e3fb976)
> 
> Tim

Hi, 
Still got the wrong message following the instruction of [#904](https://github.com/satijalab/seurat/issues/904#issuecomment-434779439) using the old version https://github.com/KlugerLab/FIt-SNE/archive/ec25f1b36598a2d21869d10a258ac366a12f0b05.zip
The version of FIt-SNE was the updated version, and the Seurat v3 was installed using the command recently:
```
install.packages('devtools')
devtools::install_github(repo = 'satijalab/seurat', ref = 'release/3.0')
library(Seurat)
```

**Comment by ChevyXu** [comment]
> > Seurat v3 was installed using the command recently
> 
> You mean _after_ the commit I mentioned? [ae9d28e](https://github.com/satijalab/seurat/commit/ae9d28e91b9bb6b750c36c3fb4783fef7e3fb976), 30th Nov
> 
> Could you share a minimal reproducible example for this issue?

Hi, @timoast 
Thanks for your kindly reply, I reinstalled the Seurat V3, and the problem solved.
The error was caused by the different version.

best
